proc-macros/src/new/render_server.rs

```diff
@@ -62,6 +61,18 @@ impl RpcDescription {
 			}
 		};
 
+		/// Helper that will ignore results of `register_*` method calls, and panic
+		/// if there have been any errors in debug builds.
+		///
+		/// The debug assert is a safeguard should the contract that guarantees the method
+		/// names to never conflict in the macro be broken in the future.
+		fn handle_register_result(tokens: TokenStream2) -> TokenStream2 {
+			quote! {{
+				let res = #tokens;
+				debug_assert!(res.is_ok(), "RPC macro method names should never conflict, this is a bug, please report it.");
```

Comment:
Why make this a debug-only assert? Isn't it worth the cost to catch these violations in all builds?

Also, I'd clarify that the bug is in the user code, in not `jsonrpsee`:
```suggestion
				debug_assert!(res.is_ok(), "RPC names must never conflict, please ensure all method names are unique.");
```

Reply:
This is just related to `jsonrpsee proc macros` that actually ensures that each method is unique so not the user should not have to do it, it would just be a compile time error then.

But this is just assertion if we introduce any bugs in that logic and I think Maciej doesn't like overhead of `expect` in release builds :P

Alternatively we could just add a test for this

Reply:
> Alternatively we could just add a test for this

How would that work? Wouldn't it have to be in the users' code?

Reply:
Yeah, it has to be in users code but we could just do a UI test:

```rust
#[rpc(client, server, namespace = "state")]
pub trait Rpc {
	/// Async method call example.
	#[method(name = "getPairs")]
	async fn storage_pairs(&self, prefix: usize, hash: Option<u128>) -> Vec<usize>;

	/// Async method call example.
	#[method(name = "getPairs")]
	async fn dup(&self, prefix: usize, hash: Option<u128>) -> Vec<usize>;
}

pub struct RpcServerImpl;

#[async_trait]
impl RpcServer for RpcServerImpl {
	async fn storage_pairs(&self, _prefix: usize, _hash: Option<u128>) -> Vec<usize> {
		vec![1, 2, 3, 4]
	}

	async fn dup(&self, _prefix: usize, _hash: Option<u128>) -> Vec<usize> {
		vec![1, 2, 3, 4]
	}
}
```

expect error: `error: "state_getPairs" is already defined`

Reply:
We already got this: https://github.com/paritytech/jsonrpsee/blob/master/proc-macros/tests/ui/incorrect/rpc/rpc_name_conflict.rs so I merged this PR